### ISSUE TITLE
Dashed Line Max Capacity

### ISSUE THREAD

**Issue by tylermanning** [tikzit/tikzit#81]
I'm really confused as to why I am getting this error whenever I use the dashed line style:
```
! TeX capacity exceeded, sorry [input stack size=5000].
<to be read again> 
                   {
l.41 		\draw [style=dashed]
                            (30.center) to (31.center);
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on preview.log.
pdflatex RETURNED AN ERROR
```
I never got it before and I've made other tikzit images using dashed lines. My style file also looks fine:
```
% TiKZ style file generated by TikZiT. You may edit this file manually,
% but some things (e.g. comments) may be overwritten. To be readable in
% TikZiT, the only non-comment lines must be of the form:
% \tikzstyle{NAME}=[PROPERTY LIST]

% Node styles
\tikzstyle{none}=[fill=none, tikzit category=nodes, shape=circle, draw=none]
\tikzstyle{red node}=[fill=red, tikzit category=nodes, shape=circle, draw=black]
\tikzstyle{small red}=[fill=red, circle, inner sep=0pt, minimum size=7pt, draw=black]
\tikzstyle{small blue}=[fill, circle, inner sep=0pt, minimum size=2pt]
\tikzstyle{blue node}=[fill=blue, shape=circle, draw=black, tikzit category=nodes]
\tikzstyle{green node}=[tikzit fill=green, fill=green, shape=circle, draw=black, tikzit category=nodes]
\tikzstyle{yellow square}=[draw=black, fill=yellow, shape=rectangle]
\tikzstyle{blue node 2}=[fill={rgb,255: red,128; green,0; blue,128}, draw=black, shape=circle, tikzit fill=blue]

% Edge styles
\tikzstyle{dashed edge}=[<->, dashed]
\tikzstyle{dashed edge2}=[->, dashed]
\tikzstyle{dashed}=[-, dashed]
\tikzstyle{blue pointer}=[->, draw=blue]
\tikzstyle{black pointer}=[->, draw=black]
\tikzstyle{black thick}=[-, draw=black, line width=0.25mm]
```

**Comment by 0xACE** [comment]
It's because one of your styles is clashing with the name `dashed`

I name my nodes `cirlceX` (with `X` as suffix for nodes)

and my edges `arrow-rightZ` (with `Z` as suffix for edges)

This way you minimze the risk of clashing file style names.

It could be addressed, but I would argue it's better that the user mitigates the issue for now by following a scheme as I proposed.

**Comment by 0xACE** [comment]
As the error implies: stack capacity has been exceeded because latex was recursively expanding:

`dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed > dashed...`

**Comment by tylermanning** [comment]
Thanks, wasn't expecting that. I renamed the styles with some suffixes as you recommended and it worked.

**Comment by 0xACE** [comment]
Also it was some time ago since I debugged the issue, but iirc it would even trigger if the name started with `dashed`, meaning if the name collides with another style it may cause issues, I also remember that `dashed arrow` would trigger this but I may be wrong.

Anyhow, the information in this thread may help with debugging the issue, especially since I described what causes the error in my follow up post...

I believe you are capable of solving it. Good luck.

I hope this didn't break your old `*.tikz` diagram as my solution requires that you walk through changing all the old files.
